c_src/adbc_nif_resource.hpp

```diff
@@ -128,14 +128,22 @@ template <typename T> struct NifRes {
 
   template <typename R = T>
   static auto destruct_resource(ErlNifEnv *env, void *args) ->
-      typename std::enable_if<!release_guard<R>::value, void>::type {}
+      typename std::enable_if<!release_guard<R>::value, void>::type {
+    if constexpr(std::is_same<struct AdbcStatement, T>::value) {
+      auto res = (NifRes<T> *)args;
+      if (res) {
+        struct AdbcError adbc_error{};
+        AdbcStatementRelease(&res->val, &adbc_error);
+      }
+    }
+  }
```

Comment:
Instead of using templates, could we define a destruct resource per type? Because we should define a release statement for each `Adbc*Release` function we have, and not only statements. WDYT? Or are templates the only way to achieve this?

Reply:
For example, the `static auto destruct_resource(ErlNifEnv *env, void *args)` above for `ArrowSchema` could have its own definition as well, right? The same for `ArrowStruct`?